wrapper/src/main/java/software/amazon/jdbc/plugin/efm/MonitorImpl.java

```diff
@@ -113,6 +113,9 @@ public MonitorImpl(
 
   @Override
   public void startMonitoring(final MonitorConnectionContext context) {
+    if (this.stopped) {
+      LOGGER.warning(() -> Messages.get("MonitorImpl.monitorIsStopped", new Object[] {this.hostSpec.getHost()}));
```

Comment:
This seems confusing, why are we logging that it is stopped. Presumably it would be stopped if we call startMonitoring?

Reply:
We're logging that the monitor is stopped in case the run method has finished and stopped is false but startMonitoring has been called again. This is because of a reported issue where the newContexts queue grew to the point of causing an out of memory error, which implies that items are being added to the queue through startMonitoring and not being removed, which would normally happen in the run method.